internal/main.go

```diff
@@ -93,14 +96,23 @@ func main() {
 
 	var dataOut []byte
 	if downtranslate {
-		// translate from 3 to 2
-		cfg, rpt, err := v3_0.Parse(dataIn)
+		// translate from 3.x to 2.2
+		cfg, rpt, err := v3_1.Parse(dataIn)
 		fmt.Fprintf(os.Stderr, "%s", rpt.String())
-		if err != nil || rpt.IsFatal() {
-			fail("Error parsing spec v3 config: %v\n%v", err, rpt)
+		if err == errors.ErrUnknownVersion {
+			// We can fail unmarshaling if it's an older config. Attempt to parse
+			// it as such.
+			cfg3_0, rpt, err := v3_0.Parse(dataIn)
+			fmt.Fprintf(os.Stderr, "%s", rpt.String())
+			if err != nil || rpt.IsFatal() {
+				fail("Error parsing spec v3.0 config: %v\n%v", err, rpt)
+			}
+			cfg = translateTo3_1.Translate(cfg3_0)
+		} else if err != nil || rpt.IsFatal() {
+			fail("Error parsing spec v3.1 config: %v\n%v", err, rpt)
 		}
 
-		newCfg, err := ignconverter.Translate3_0to2_2(cfg)
+		newCfg, err := ignconverter.Translate3_1to2_2(cfg)
```

Comment:
How is the command-line tool expected to be used?  Would it make sense to translate to 2.4, since that conversion is not as lossy?

Reply:
agreed. Changed it accordingly